File: java/src/main/java/com/creolecentric/api/CreoleCentricAPI.java
package com.creolecentric.api;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.annotations.SerializedName;
import com.google.gson.reflect.TypeToken;

import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;
import java.util.List;
import java.util.Map;

/**
 * CreoleCentric TTS API Client
 *
 * A Java client for the CreoleCentric Text-to-Speech API.
 * Requires Java 11+ for HttpClient support.
 *
 * Example usage:
 * <pre>
 * CreoleCentricAPI client = new CreoleCentricAPI("cc_your_api_key_here");
 * HealthCheckResponse health = client.checkHealth();
 * System.out.println("API Version: " + health.version);
 * </pre>
 */
public class CreoleCentricAPI {
    private final String apiKey;
    private final String baseUrl;
    private final HttpClient httpClient;
    private final Gson gson;

    // Response classes
    public static class HealthCheckResponse {
        public String status;
        public String service;
        public String database;
        public String version;
    }

    public static class CreditBalanceResponse {
        @SerializedName("total_credits")
        public double totalCredits;

        @SerializedName("used_credits")
        public double usedCredits;

        @SerializedName("available_credits")
        public double availableCredits;

        public String currency;
    }

    public static class Voice {
        @SerializedName("voice_id")
        public String voiceId;

        public String name;
        public String region;
        public String gender;
        public String description;

        @SerializedName("preview_url")
        public String previewUrl;

        @SerializedName("is_active")
        public boolean isActive;
    }

    public static class VoicesResponse {
        public boolean success;
        public List<Voice> voices;
        public int count;
        public String source;
    }

    public static class Model {
        public String id;
        public String name;
        public String description;
        public String version;

        @SerializedName("is_active")
        public boolean isActive;

        @SerializedName("supports_streaming")
        public boolean supportsStreaming;
    }

    public static class ModelsResponse {
        public boolean success;
        public List<Model> models;
        public int count;
    }

    public static class TTSJob {
        public String id;
        public String status;

        @SerializedName("text_input")
        public String textInput;

        @SerializedName("voice_id")
        public String voiceId;

        @SerializedName("model_id")
        public String modelId;

        @SerializedName("created_at")
        public String createdAt;
    }

    public static class JobStatusResponse {
        public String id;
        public String status;

        @SerializedName("audio_url")
        public String audioUrl;

        @SerializedName("created_at")
        public String createdAt;

        @SerializedName("updated_at")
        public String updatedAt;

        public String error;
    }

    public static class RecentJobsResponse {
        public boolean success;
        public List<JobStatusResponse> jobs;
        public int count;
    }

    /**
     * Initialize the CreoleCentric API client
     *
     * @param apiKey Your CreoleCentric API key (starts with 'cc_')
     */
    public CreoleCentricAPI(String apiKey) {
        this(apiKey, "https://creolecentric.com/api/v1");
    }

    /**
     * Initialize the CreoleCentric API client with custom base URL
     *
     * @param apiKey Your CreoleCentric API key
     * @param baseUrl Custom base URL for the API
     */
    public CreoleCentricAPI(String apiKey, String baseUrl) {
        this.apiKey = apiKey;
        this.baseUrl = baseUrl;
        this.httpClient = HttpClient.newBuilder()
                .connectTimeout(Duration.ofSeconds(30))
                .build();
        this.gson = new GsonBuilder().create();
    }

    /**
     * Make an HTTP request to the API
     */
    private <T> T makeRequest(String method, String endpoint, Object body, Class<T> responseClass) throws IOException, InterruptedException {
        String url = baseUrl + endpoint;

        HttpRequest.Builder requestBuilder = HttpRequest.newBuilder()
                .uri(URI.create(url))
                .timeout(Duration.ofSeconds(30))
                .header("Authorization", "Bearer " + apiKey)
                .header("Content-Type", "application/json");

        if ("POST".equals(method) && body != null) {
            String jsonBody = gson.toJson(body);
            requestBuilder.POST(HttpRequest.BodyPublishers.ofString(jsonBody));
        } else {
            requestBuilder.GET();
        }

        HttpRequest request = requestBuilder.build();
        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());

        if (response.statusCode() >= 400) {
            throw new IOException("HTTP " + response.statusCode() + ": " + response.body());
        }

        return gson.fromJson(response.body(), responseClass);
    }

    /**
     * Check API health and get version information
     */
    public HealthCheckResponse checkHealth() throws IOException, InterruptedException {
        return makeRequest("GET", "/health/", null, HealthCheckResponse.class);
    }

    /**
     * Get current credit balance
     */
    public CreditBalanceResponse getCreditBalance() throws IOException, InterruptedException {
        return makeRequest("GET", "/credits/balance/", null, CreditBalanceResponse.class);
    }

    /**
     * Get list of available voices
     */
    public VoicesResponse getVoices() throws IOException, InterruptedException {
        return makeRequest("GET", "/tts/voices/", null, VoicesResponse.class);
    }

    /**
     * Get list of available models
     */
    public ModelsResponse getModels() throws IOException, InterruptedException {
        return makeRequest("GET", "/tts/models/", null, ModelsResponse.class);
    }

    /**
     * Create a new TTS job
     *
     * @param text The text to convert to speech
     * @param voiceId The ID of the voice to use
     * @param modelId The ID of the model to use
     * @return The created job
     */
    public TTSJob createTTSJob(String text, String voiceId, String modelId) throws IOException, InterruptedException {
        return createTTSJob(text, voiceId, modelId, 1.0, 0.5, 0.75);
    }

    /**
     * Create a new TTS job with custom parameters
     *
     * @param text The text to convert to speech
     * @param voiceId The ID of the voice to use
     * @param modelId The ID of the model to use
     * @param speed Speech speed (0.5 to 2.0)
     * @param stability Voice stability (0.0 to 1.0)
     * @param similarityBoost Voice similarity boost (0.0 to 1.0)
     * @return The created job
     */
    public TTSJob createTTSJob(String text, String voiceId, String modelId,
                               double speed, double stability, double similarityBoost) throws IOException, InterruptedException {
        Map<String, Object> payload = Map.of(
            "text", text,
            "voice_id", voiceId,
            "model_id", modelId,
            "speed", speed,
            "stability", stability,
            "similarity_boost", similarityBoost
        );

        return makeRequest("POST", "/tts/submit/", payload, TTSJob.class);
    }

    /**
     * Get the status of a TTS job
     *
     * @param jobId The job ID
     * @return The job status
     */
    public JobStatusResponse getJobStatus(String jobId) throws IOException, InterruptedException {
        return makeRequest("GET", "/tts/job/" + jobId + "/", null, JobStatusResponse.class);
    }

    /**
     * Wait for a job to complete
     *
     * @param jobId The job ID
     * @param timeout Maximum time to wait in seconds
     * @param pollInterval How often to check status in seconds
     * @return The final job status
     */
    public JobStatusResponse waitForJob(String jobId, int timeout, int pollInterval) throws IOException, InterruptedException {
        long startTime = System.currentTimeMillis();
        long timeoutMs = timeout * 1000L;
        long pollIntervalMs = pollInterval * 1000L;

        while (System.currentTimeMillis() - startTime < timeoutMs) {
            JobStatusResponse status = getJobStatus(jobId);

            if ("completed".equals(status.status) ||
                "delivered".equals(status.status) ||
                "failed".equals(status.status) ||
                "cancelled".equals(status.status)) {
                return status;
            }

            System.out.println("Job " + jobId + " status: " + status.status);
            Thread.sleep(pollIntervalMs);
        }

        throw new IOException("Job " + jobId + " did not complete within " + timeout + " seconds");
    }

    /**
     * Wait for a job to complete with default timeout (300s) and poll interval (2s)
     */
    public JobStatusResponse waitForJob(String jobId) throws IOException, InterruptedException {
        return waitForJob(jobId, 300, 2);
    }

    /**
     * Get recent TTS jobs
     *
     * @param limit Maximum number of jobs to return
     * @return List of recent jobs
     */
    public RecentJobsResponse getRecentJobs(int limit) throws IOException, InterruptedException {
        return makeRequest("GET", "/tts/jobs/?limit=" + limit, null, RecentJobsResponse.class);
    }

    /**
     * Get recent TTS jobs (default limit: 10)
     */
    public RecentJobsResponse getRecentJobs() throws IOException, InterruptedException {
        return getRecentJobs(10);
    }

    /**
     * Example usage demonstrating all features
     */
    public static void main(String[] args) {
        String apiKey = System.getenv("CREOLECENTRIC_API_KEY");
        if (apiKey == null || apiKey.isEmpty()) {
            System.err.println("Error: CREOLECENTRIC_API_KEY environment variable not set");
            System.exit(1);
        }

        try {
            CreoleCentricAPI client = new CreoleCentricAPI(apiKey);

            // 1. Health Check
            System.out.println("=== Health Check ===");
            HealthCheckResponse health = client.checkHealth();
            System.out.println("Status: " + health.status);
            System.out.println("Service: " + health.service);
            System.out.println("Version: " + health.version);
            System.out.println();

            // 2. Credit Balance
            System.out.println("=== Credit Balance ===");
            CreditBalanceResponse balance = client.getCreditBalance();
            System.out.println("Total Credits: " + balance.totalCredits);
            System.out.println("Used Credits: " + balance.usedCredits);
            System.out.println("Available Credits: " + balance.availableCredits);
            System.out.println();

            // 3. List Voices
            System.out.println("=== Available Voices ===");
            VoicesResponse voicesResponse = client.getVoices();
            List<Voice> voices = voicesResponse.voices;
            System.out.println("Found " + voicesResponse.count + " voices (source: " + voicesResponse.source + "):");
            for (int i = 0; i < Math.min(5, voices.size()); i++) {
                Voice voice = voices.get(i);
                System.out.println("  - " + voice.name + " (" + voice.voiceId + ")");
                System.out.println("    Region: " + voice.region + ", Gender: " + voice.gender);
            }
            System.out.println();

            // 4. List Models
            System.out.println("=== Available Models ===");
            ModelsResponse modelsResponse = client.getModels();
            List<Model> models = modelsResponse.models;
            System.out.println("Found " + modelsResponse.count + " models:");
            for (Model model : models) {
                System.out.println("  - " + model.name + " (" + model.id + ")");
                System.out.println("    " + model.description);
            }
            System.out.println();

            // 5. Create TTS Job
            String voiceId = "i4mRPwKM2yHwXhbmkN514";  // Xavier Bruneau
            String modelId = "ccl_ht_v100";

            if (!voices.isEmpty() &&
                !"voice_1".equals(voices.get(0).voiceId) &&
                !"voice_2".equals(voices.get(0).voiceId)) {
                voiceId = voices.get(0).voiceId;
            }
            if (!models.isEmpty()) {
                modelId = models.get(0).id;
            }

            System.out.println("=== Creating TTS Job ===");
            String text = "Bonjou! Sa se yon egzanp nan itilizasyon API CreoleCentric pou konvèti tèks an Kreyòl Ayisyen an lapawòl.";
            System.out.println("Text: " + text);
            System.out.println("Voice ID: " + voiceId);
            System.out.println("Model ID: " + modelId);

            TTSJob job = client.createTTSJob(text, voiceId, modelId);
            System.out.println("Job created: " + job.id);
            System.out.println("Status: " + job.status);
            System.out.println();

            // 6. Wait for Completion
            System.out.println("=== Waiting for Job Completion ===");
            JobStatusResponse finalStatus = client.waitForJob(job.id, 60, 2);
            System.out.println("Job " + finalStatus.id + " " + finalStatus.status);

            if ("completed".equals(finalStatus.status) || "delivered".equals(finalStatus.status)) {
                System.out.println("Audio URL: " + finalStatus.audioUrl);
            } else if ("failed".equals(finalStatus.status)) {
                System.out.println("Error: " + finalStatus.error);
            }
            System.out.println();

            // 7. Recent Jobs
            System.out.println("=== Recent Jobs ===");
            RecentJobsResponse recentJobs = client.getRecentJobs(5);
            System.out.println("Found " + recentJobs.count + " recent jobs:");
            for (JobStatusResponse recentJob : recentJobs.jobs) {
                System.out.println("  - Job " + recentJob.id + ": " + recentJob.status);
                if (recentJob.audioUrl != null) {
                    System.out.println("    Audio: " + recentJob.audioUrl);
                }
            }

        } catch (IOException | InterruptedException e) {
            System.err.println("Error: " + e.getMessage());
            e.printStackTrace();
            System.exit(1);
        }
    }
}

File: go/basic-example/creolecentric_api.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"os"
	"strings"
	"time"

	"github.com/joho/godotenv"
)

// ============== Type Definitions ==============

type HealthResponse struct {
	Status   string `json:"status"`
	Service  string `json:"service"`
	Database string `json:"database"`
	Version  string `json:"version"`
}

type CreditBalance struct {
	TotalCredits        int    `json:"total_credits"`
	SubscriptionCredits int    `json:"subscription_credits"`
	PurchasedCredits    int    `json:"purchased_credits"`
	SubscriptionPlan    string `json:"subscription_plan,omitempty"`
}

type Voice struct {
	ID                    string   `json:"id"`
	VoiceID               string   `json:"voice_id"`
	Name                  string   `json:"name"`
	Description           string   `json:"description"`
	Gender                string   `json:"gender"`
	Region                string   `json:"region"`
	RegionCode            string   `json:"region_code,omitempty"`
	Accent                string   `json:"accent,omitempty"`
	Style                 string   `json:"style,omitempty"`
	QualityLevel          string   `json:"quality_level,omitempty"`
	AgeGroup              string   `json:"age_group,omitempty"`
	LanguageCode          string   `json:"language_code"`
	IsActive              bool     `json:"is_active"`
	IsDefault             bool     `json:"is_default"`
	IsFeatured            bool     `json:"is_featured"`
	IsPremium             bool     `json:"is_premium"`
	PreviewURL            string   `json:"preview_url,omitempty"`
	PreviewText           string   `json:"preview_text,omitempty"`
	Tags                  []string `json:"tags,omitempty"`
	DefaultStability      float64  `json:"default_stability,omitempty"`
	DefaultSimilarityBoost float64 `json:"default_similarity_boost,omitempty"`
}

type VoicesResponse struct {
	Success bool    `json:"success"`
	Voices  []Voice `json:"voices"`
	Count   int     `json:"count"`
	Source  string  `json:"source"`
}

type Model struct {
	ID                 string   `json:"id"`
	Name               string   `json:"name"`
	DisplayName        string   `json:"display_name,omitempty"`
	Description        string   `json:"description"`
	Version            string   `json:"version,omitempty"`
	QualityScore       float64  `json:"quality_score,omitempty"`
	IsDefault          bool     `json:"is_default,omitempty"`
	IsPremium          bool     `json:"is_premium,omitempty"`
	SupportedLanguages []string `json:"supported_languages,omitempty"`
}

type ModelsResponse struct {
	Success bool    `json:"success"`
	Models  []Model `json:"models"`
	Count   int     `json:"count"`
}

type TTSJob struct {
	ID                  string  `json:"id"`
	Text                string  `json:"text"`
	VoiceID             string  `json:"voice_id"`
	VoiceName           string  `json:"voice_name,omitempty"`
	ModelID             string  `json:"model_id"`
	Status              string  `json:"status"`
	CreatedAt           string  `json:"created_at"`
	StartedAt           string  `json:"started_at,omitempty"`
	CompletedAt         string  `json:"completed_at,omitempty"`
	AudioFileURL        string  `json:"audio_file_url,omitempty"`
	DurationSeconds     float64 `json:"duration_seconds,omitempty"`
	DurationFormatted   string  `json:"duration_formatted,omitempty"`
	CreditsUsed         int     `json:"credits_used"`
	CharactersProcessed int     `json:"characters_processed,omitempty"`
	ErrorMessage        string  `json:"error_message,omitempty"`
}

type JobStatusResponse struct {
	Status          string  `json:"status"`
	AudioURL        string  `json:"audio_url,omitempty"`
	DurationSeconds float64 `json:"duration_seconds,omitempty"`
	ErrorMessage    string  `json:"error_message,omitempty"`
}

type JobListResponse struct {
	Results  []TTSJob `json:"results"`
	Count    int      `json:"count"`
	Next     string   `json:"next,omitempty"`
	Previous string   `json:"previous,omitempty"`
}

// ============== API Client ==============

type CreoleCentricAPI struct {
	APIKey  string
	BaseURL string
	Client  *http.Client
}

func NewCreoleCentricAPI(apiKey string, baseURL ...string) *CreoleCentricAPI {
	url := "https://api.creolecentric.com/v1"
	if len(baseURL) > 0 {
		url = strings.TrimSuffix(baseURL[0], "/")
	}

	return &CreoleCentricAPI{
		APIKey:  apiKey,
		BaseURL: url,
		Client: &http.Client{
			Timeout: 30 * time.Second,
		},
	}
}

func (api *CreoleCentricAPI) makeRequest(method, endpoint string, body interface{}) ([]byte, error) {
	url := api.BaseURL + endpoint

	var reqBody io.Reader
	if body != nil {
		jsonData, err := json.Marshal(body)
		if err != nil {
			return nil, fmt.Errorf("failed to marshal request body: %w", err)
		}
		reqBody = bytes.NewBuffer(jsonData)
	}

	req, err := http.NewRequest(method, url, reqBody)
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}

	req.Header.Set("Authorization", fmt.Sprintf("ApiKey %s", api.APIKey))
	req.Header.Set("Content-Type", "application/json")

	resp, err := api.Client.Do(req)
	if err != nil {
		return nil, fmt.Errorf("request failed: %w", err)
	}
	defer resp.Body.Close()

	data, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, fmt.Errorf("failed to read response body: %w", err)
	}

	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
		return nil, fmt.Errorf("HTTP %d: %s", resp.StatusCode, string(data))
	}

	return data, nil
}

// ============== Health Check ==============

func (api *CreoleCentricAPI) CheckHealth() (*HealthResponse, error) {
	data, err := api.makeRequest("GET", "/health/", nil)
	if err != nil {
		return nil, err
	}

	var health HealthResponse
	if err := json.Unmarshal(data, &health); err != nil {
		return nil, fmt.Errorf("failed to parse health response: %w", err)
	}

	return &health, nil
}

// ============== User & Credits ==============

func (api *CreoleCentricAPI) GetCreditBalance() (*CreditBalance, error) {
	data, err := api.makeRequest("GET", "/credits/balance/", nil)
	if err != nil {
		return nil, err
	}

	var balance CreditBalance
	if err := json.Unmarshal(data, &balance); err != nil {
		return nil, fmt.Errorf("failed to parse credit balance: %w", err)
	}

	return &balance, nil
}

// ============== Voices & Models ==============

func (api *CreoleCentricAPI) GetVoices() (*VoicesResponse, error) {
	data, err := api.makeRequest("GET", "/tts/voices/", nil)
	if err != nil {
		return nil, err
	}

	var voicesResp VoicesResponse
	if err := json.Unmarshal(data, &voicesResp); err != nil {
		return nil, fmt.Errorf("failed to parse voices response: %w", err)
	}

	return &voicesResp, nil
}

func (api *CreoleCentricAPI) GetModels() (*ModelsResponse, error) {
	data, err := api.makeRequest("GET", "/tts/models/", nil)
	if err != nil {
		return nil, err
	}

	var modelsResp ModelsResponse
	if err := json.Unmarshal(data, &modelsResp); err != nil {
		return nil, fmt.Errorf("failed to parse models response: %w", err)
	}

	return &modelsResp, nil
}

// ============== TTS Jobs ==============

type CreateJobRequest struct {
	Text               string  `json:"text"`
	VoiceID            string  `json:"voice_id"`
	ModelID            string  `json:"model_id"`
	WebhookURL         string  `json:"webhook_url,omitempty"`
	Speed              float64 `json:"speed,omitempty"`
	Stability          float64 `json:"stability,omitempty"`
	SimilarityBoost    float64 `json:"similarity_boost,omitempty"`
	Style              float64 `json:"style,omitempty"`
	UseSpeakerBoost    bool    `json:"use_speaker_boost,omitempty"`
}

func (api *CreoleCentricAPI) CreateTTSJob(text, voiceID, modelID string) (*TTSJob, error) {
	reqBody := CreateJobRequest{
		Text:    text,
		VoiceID: voiceID,
		ModelID: modelID,
	}

	data, err := api.makeRequest("POST", "/tts/jobs/", reqBody)
	if err != nil {
		return nil, err
	}

	var job TTSJob
	if err := json.Unmarshal(data, &job); err != nil {
		return nil, fmt.Errorf("failed to parse job response: %w", err)
	}

	return &job, nil
}

func (api *CreoleCentricAPI) GetJobStatus(jobID string) (*JobStatusResponse, error) {
	data, err := api.makeRequest("GET", fmt.Sprintf("/tts/jobs/%s/status/", jobID), nil)
	if err != nil {
		return nil, err
	}

	var status JobStatusResponse
	if err := json.Unmarshal(data, &status); err != nil {
		return nil, fmt.Errorf("failed to parse job status: %w", err)
	}

	return &status, nil
}

func (api *CreoleCentricAPI) ListJobs(limit int) (*JobListResponse, error) {
	endpoint := fmt.Sprintf("/tts/jobs/list/?limit=%d", limit)
	data, err := api.makeRequest("GET", endpoint, nil)
	if err != nil {
		return nil, err
	}

	var jobList JobListResponse
	if err := json.Unmarshal(data, &jobList); err != nil {
		return nil, fmt.Errorf("failed to parse job list: %w", err)
	}

	return &jobList, nil
}


// ============== Example Usage ==============

func main() {
	// Load environment variables
	godotenv.Load()

	apiKey := os.Getenv("CREOLECENTRIC_API_KEY")
	if apiKey == "" {
		fmt.Println("Error: CREOLECENTRIC_API_KEY environment variable not set")
		fmt.Println("Please set it with your API key: export CREOLECENTRIC_API_KEY='cc_your_key_here'")
		os.Exit(1)
	}

	client := NewCreoleCentricAPI(apiKey)

	// 1. Check API health
	fmt.Println(strings.Repeat("=", 50))
	fmt.Println("1. Checking API Health")
	fmt.Println(strings.Repeat("=", 50))
	health, err := client.CheckHealth()
	if err != nil {
		fmt.Printf("Error: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("API Status: %s\n", health.Status)
	fmt.Printf("Version: %s\n\n", health.Version)

	// 2. Get credit balance
	fmt.Println(strings.Repeat("=", 50))
	fmt.Println("2. Credit Balance")
	fmt.Println(strings.Repeat("=", 50))
	balance, err := client.GetCreditBalance()
	if err != nil {
		fmt.Printf("Error: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("Total Credits: %d\n", balance.TotalCredits)
	fmt.Printf("Subscription Credits: %d\n", balance.SubscriptionCredits)
	fmt.Printf("Purchased Credits: %d\n\n", balance.PurchasedCredits)

	// 3. Get available voices
	fmt.Println(strings.Repeat("=", 50))
	fmt.Println("3. Available Voices")
	fmt.Println(strings.Repeat("=", 50))
	voicesResp, err := client.GetVoices()
	if err != nil {
		fmt.Printf("Error: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("Found %d voices (source: %s):\n", voicesResp.Count, voicesResp.Source)
	for i, voice := range voicesResp.Voices {
		if i >= 5 {
			break
		}
		fmt.Printf("  - %s (ID: %s)\n", voice.Name, voice.VoiceID)
		fmt.Printf("    Region: %s, Gender: %s\n", voice.Region, voice.Gender)
	}
	fmt.Println()

	// 4. Get available models
	fmt.Println(strings.Repeat("=", 50))
	fmt.Println("4. Available Models")
	fmt.Println(strings.Repeat("=", 50))
	modelsResp, err := client.GetModels()
	if err != nil {
		fmt.Printf("Error: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("Found %d models:\n", modelsResp.Count)
	for _, model := range modelsResp.Models {
		displayName := model.DisplayName
		if displayName == "" {
			displayName = model.Name
		}
		fmt.Printf("  - %s (ID: %s)\n", displayName, model.ID)
		fmt.Printf("    Description: %s\n", model.Description)
	}
	fmt.Println()

	// 5. Create a TTS job
	fmt.Println(strings.Repeat("=", 50))
	fmt.Println("5. Creating TTS Job")
	fmt.Println(strings.Repeat("=", 50))

	text := "Bonjou! Mwen se yon egzanp API pou CreoleCentric. Mwen ka pale Kreyòl ayisyen."
	fmt.Printf("Text: %s\n", text)

	voiceID := "i4mRPwKM2yHwXhbmkN514" // Xavier Bruneau
	modelID := "ccl_ht_v100"

	if len(voicesResp.Voices) > 0 &&
	   voicesResp.Voices[0].VoiceID != "voice_1" &&
	   voicesResp.Voices[0].VoiceID != "voice_2" {
		voiceID = voicesResp.Voices[0].VoiceID
	}
	if len(modelsResp.Models) > 0 {
		modelID = modelsResp.Models[0].ID
	}

	job, err := client.CreateTTSJob(text, voiceID, modelID)
	if err != nil {
		fmt.Printf("Error: %v\n", err)
		os.Exit(1)
	}

	fmt.Println("Job created successfully!")
	fmt.Printf("Job ID: %s\n", job.ID)
	fmt.Printf("Status: %s\n", job.Status)
	fmt.Printf("Credits used: %d\n\n", job.CreditsUsed)
	fmt.Println("📢 To receive webhook notifications, add webhook_url to CreateJobRequest:")
	fmt.Println("   Events: tts_queued → tts_started → tts_synthesized → tts_uploaded → tts_delivered")
	fmt.Println("   See examples/webhook_server.go for webhook handling example")
	fmt.Println()

	// 6. List recent jobs
	fmt.Println(strings.Repeat("=", 50))
	fmt.Println("6. Recent Jobs")
	fmt.Println(strings.Repeat("=", 50))

	jobList, err := client.ListJobs(5)
	if err != nil {
		fmt.Printf("Error: %v\n", err)
		os.Exit(1)
	}

	fmt.Printf("Recent %d jobs:\n", len(jobList.Results))
	for _, job := range jobList.Results {
		jobIDDisplay := job.ID
		if len(job.ID) > 8 {
			jobIDDisplay = job.ID[:8] + "..."
		}
		fmt.Printf("  - Job %s\n", jobIDDisplay)
		fmt.Printf("    Created: %s\n", job.CreatedAt)
		fmt.Printf("    Status: %s\n", job.Status)
		if job.Text != "" {
			textPreview := job.Text
			if len(textPreview) > 50 {
				textPreview = textPreview[:50] + "..."
			}
			fmt.Printf("    Text: %s\n", textPreview)
		}
	}
}
